client/client.go

```diff
@@ -46,328 +51,439 @@ func New_Old(nc *nats.Conn, prefix string, timeout time.Duration) Client {
 	if err != nil {
 		panic(err)
 	}
-	return c.printResults(c.nc, subject, &b_data, &timeout)
+	acks := []*models.ProviderAuctionAck{}
+	rawAcks := c.CollectTimeout(subject, &data_bytes, &timeout)
+	for _, a := range rawAcks {
+		pAck := models.ProviderAuctionAck{}
+		err := json.Unmarshal([]byte(a), &pAck)
+		if err != nil {
+			return nil, err
+		}
+		acks = append(acks, &pAck)
+	}
+	return acks, nil
 }
 
 // NATs topic: cmd.{host}.la
-func (c Client) StartActor(hostID string, actorRef string, count int, annotations map[string]string) []string {
+func (c Client) StartActor(hostID string, actorRef string, count uint16, annotations map[string]string) (*models.CtlOperationAck, error) {
 	subject := broker.Commands{}.StartActor(c.nsPrefix, hostID)
 	log.Debug(subject)
-	data := struct {
-		ActorRef    string            `json:"actor_ref"`
-		HostID      string            `json:"host_id"`
-		Count       int               `json:"count"`
-		Annotations map[string]string `json:"annotations,omitempty"`
-	}{
+	data := models.StartActorCommand{
 		ActorRef:    actorRef,
-		HostID:      hostID,
+		HostId:      hostID,
 		Count:       count,
 		Annotations: annotations,
 	}
 
-	b_data, err := json.Marshal(data)
+	data_bytes, err := json.Marshal(data)
 	if err != nil {
-		panic(err)
+		return nil, err
 	}
 
-	return c.printResults(c.nc, subject, &b_data, nil)
+	rawAcks := c.CollectTimeout(subject, &data_bytes, nil)
+	for _, a := range rawAcks {
+		cAck := models.CtlOperationAck{}
+		err := json.Unmarshal([]byte(a), &cAck)
+		if err != nil {
+			return nil, err
+		}
+		if cAck.Accepted {
+			return &cAck, nil
+		}
+	}
+	return nil, errors.New("did not receive ack")
 }
 
 // NATs topic: cmd.{host}.lp
-func (c Client) StartProvider(hostID string, providerRef string, linkName string, annotations map[string]string, providerConfiguration string) []string {
+func (c Client) StartProvider(hostID string, providerRef string, linkName string, annotations map[string]string, providerConfiguration string) (*models.CtlOperationAck, error) {
 	subject := broker.Commands{}.StartProvider(c.nsPrefix, hostID)
-	log.Debug(subject)
-	data := struct {
-		ProviderRef   string            `json:"provider_ref"`
-		HostID        string            `json:"host_id"`
-		LinkName      string            `json:"link_name"`
-		Annotations   map[string]string `json:"annotations,omitempty"`
-		Configuration string            `json:"configuration"`
-	}{
+	startCmd := models.StartProviderCommand{
 		ProviderRef:   providerRef,
-		HostID:        hostID,
+		HostId:        hostID,
 		LinkName:      linkName,
 		Annotations:   annotations,
 		Configuration: providerConfiguration,
 	}
 
-	b_data, err := json.Marshal(data)
+	if hostID == "" {
+		subject := broker.ProviderAuctionSubject(c.nsPrefix)
+		proReq := models.ProviderAuctionRequest{
+			ProviderRef: providerRef,
+			LinkName:    linkName,
+			// TODO: where are these contrainsts??
+			Constraints: models.ConstraintMap{},
```

Comment:
So, is is _suppose_ to be auctioning?? I ask because of this comment https://github.com/wasmCloud/control-interface-client/blob/1da36e3b146e8315754fc8a60617d839e90c0df0/src/lib.rs#L529

Reply:
The comment says one thing, but the code says another 😅

Reply:
I created an issue for this